### PR TITLE
[clapack] Fix linux build

### PR DESCRIPTION
**Pull request by kafeg**
Fix for https://github.com/Microsoft/vcpkg/issues/5368

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@kafeg, thanks for contributing to vcpkg! 

Here is fail triplet(x64-windows-static) and regression(openblas) with your PR.

Log:
 Processing x64-windows-static                   757 vs 767
```
    blaze                                                        Skip vs Fail
     ceres                                                       Skip vs Pass
     clapack                                                     Skip vs Pass
     dlib                                                        Skip vs Pass
     ensmallen                                                   Skip vs Pass
     g2o                                                         Skip vs Pass
     geogram                                                     Skip vs Pass
     mlpack                                                      Skip vs Pass
     openblas                      **regression**                Fail vs Pass
     openmvg                                                     Skip vs Pass
     sophus                                                      Skip vs Fail
     suitesparse                                                 Skip vs Pass
```

File: ports/openblas/portfile.cmake [hunk 1]
```diff
@@ -16,9 +16,11 @@ if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
     message(FATAL_ERROR "openblas can only be built for x64 currently")
 endif()
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    message("openblas currenly only supports dynamic library linkage")
-    set(VCPKG_LIBRARY_LINKAGE "dynamic")
+if(VCPKG_CMAKE_SYSTEM_NAME  STREQUAL "WindowsStore")
```
Comment (seanwarren) [inline ports/openblas/portfile.cmake]: This check will only trigger on UWP builds, not normal windows builds (as is currently required to build `OpenBLAS` successfully). I would suggest the following change which matches the `OpenBLAS` restriction

```suggestion
if(MSVC)
```

**Comment by PhoebeHui** [comment]
@seanwarren, could you also update CONTROL revision in https://github.com/Microsoft/vcpkg/blob/master/ports/clapack/CONTROL and 
https://github.com/Microsoft/vcpkg/blob/master/ports/openblas/CONTROL? 

More information in https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/control-files.md

**Comment by seanwarren** [comment]
@PhoebeHui - this isn't my PR so I don't have write access (I just made a code review suggestion earlier). I think @kafeg will need to update the control file. 

Cheers

**Comment by vicroms** [comment]
It is failing due to a post-build validation:

```
DLLs should not be present in a static build, but the following DLLs were found:

    C:/vsts/_work/1/s/packages/openblas_x64-windows-static/bin/openblas.dll
    C:/vsts/_work/1/s/packages/openblas_x64-windows-static/debug/bin/openblas.dll

There should be no bin\ directory in a static build, but ...\packages\openblas_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but ...\packages\openblas_x64-windows-static\debug\bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
    endif()

Found 2 error(s). Please correct the portfile:
```

**Comment by grdowns** [comment]
Thanks for the contributions, everyone!

**Pull request merged**